### DISCUSSION TITLE
How to change concurrent (parallel) jobs number?

### DISCUSSION THREAD

**Discussion by mittalyashu** [community/community#26586] [Actions]
<p>I have created a workflow, where I want it should run 1 at a time instead of running in parallel.</p>
<p>BTW, I am talking about these concurrent jobs<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/9/91e3fdaeb04cf4803c2aeb2513456f19e0a786ac.png" data-download-href="/uploads/short-url/kOBHuOMbnD7BgBBa3ZmtDI4RVec.png?dl=1" title="Screenshot_2020-09-13_at_7_46_14_PM" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/29014463/181098586-69c118b9-6d4c-48ca-a4e7-f37c406d0c08.png" alt="Screenshot_2020-09-13_at_7_46_14_PM" data-base62-sha1="kOBHuOMbnD7BgBBa3ZmtDI4RVec" width="517" height="165" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/9/91e3fdaeb04cf4803c2aeb2513456f19e0a786ac_2_517x165.png, https://aws1.discourse-cdn.com/github/optimized/2X/9/91e3fdaeb04cf4803c2aeb2513456f19e0a786ac_2_775x247.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/9/91e3fdaeb04cf4803c2aeb2513456f19e0a786ac_2_1034x330.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/9/91e3fdaeb04cf4803c2aeb2513456f19e0a786ac_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot_2020-09-13_at_7_46_14_PM</span><span class="informations">1858×594 30.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Not these<br>
<img src="https://user-images.githubusercontent.com/29014463/181098587-3d3f4fb8-7492-4fc3-acc5-9a9d9c9d818f.png" alt="Screenshot 2020-09-13 at 7.45.17 PM" data-base62-sha1="pRd704eTZVSNhyw9ODxnN0BTFUG" width="343" height="225"></p>

**Comment by shrink** [comment]
<p>A workflow has jobs, a job has steps. You’re describing the concurrency of <em>workflows</em>, not jobs – minor distinction in language but important when thinking about the documentation and how you solve this problem.</p>
<p>GitHub Actions do not currently support managing workflow concurrency but there is an action called <a href="https://github.com/softprops/turnstyle" rel="nofollow noopener">turnstyle</a> which will block workflows, meaning that only one can run at a time. This is less than perfect as a solution (because you’ll be charged for the waiting job minutes) but it will allow for your use-case of enforcing a concurrency limit of 1 for workflows.</p>
<p>That said, if you do have a dependency between workflows, it is a sign that you <em>may</em> wish to reconsider whether multiple workflows is appropriate and if instead it should be dependent jobs (using <code>needs</code>).</p>

**Comment by mittalyashu** [comment]
<p>Ah! Sorry for incorrect terminology. I wasn’t sure how to actually say it.</p>
<p>Anyway, using <strong>turnstyle</strong> might fulfil my use-case for now.</p>
<p>But as it will cost even if the other workflows are waiting in queue.</p>
<blockquote>
<p>If GitHub Staff is watching this post, please consider adding the feature to defining how many concurrent workflow it can run.</p>
<p>Just a side note: it might also help who are using self-hosted runners while there server might not be able to handle multiple running workflows at the same time <em>(for certain use-cases)</em>, but with <code>parallel-workflows: 2</code> <em>(something like this)</em>, it will put their servers at it ease <em>(hope that make sense)</em>.</p>
</blockquote>

**Comment by shrink** [comment]
<p>I did some further digging and you could use the new <a href="https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/" rel="nofollow noopener"><code>workflow_dispatch</code></a> event to meet your needs without incurring additional minutes. Although not perfect because it will tightly couple each of your workflows, use <a href="https://github.com/marketplace/actions/workflow-dispatch" rel="nofollow noopener">benc-uk/workflow-dispatch@v1</a> to chain the workflows together.</p>
<p>Set your first workflow (e.g: <code>Workflow A</code>) to run on push, then as the final step in the final job of Workflow A you would publish a <code>workflow_dispatch</code> event for the next workflow which would be set to be triggered by <code>workflow_dispatch</code> instead of <code>push</code>. Repeat for each subsequent Workflow in the chain.</p>
<pre><code class="lang-auto">name: Workflow A

on: [push]

jobs:
  first-job:
    runs-on: ubuntu-latest
    steps:
      - run: echo "Hello, World!"
      - uses: benc-uk/workflow-dispatch@v1
         with:
           workflow: Workflow B
           token: ${{ secrets.PERSONAL_TOKEN }}
</code></pre>
<pre><code class="lang-auto">name: Workflow B

on: [workflow_dispatch]

jobs:
  first-job:
    runs-on: ubuntu-latest
    steps:
      - run: echo "Hello, World!"
      - uses: benc-uk/workflow-dispatch@v1
         with:
           workflow: Workflow C
           token: ${{ secrets.PERSONAL_TOKEN }}
</code></pre>

**Comment by Simran-B** [comment, marked as answer]
<p>There’s a new feature that you can use to limit the concurrency of workflows:</p>
<aside class="quote" data-post="44" data-topic="16844">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/how-to-limit-concurrent-workflow-runs/16844/44">How to limit concurrent workflow runs</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    We have recently shipped a feature that will enable you to configure workflow level concurrency for some scenarios <a href="https://github.blog/changelog/2021-04-19-github-actions-limit-workflow-run-or-job-concurrency/" rel="noopener nofollow ugc">GitHub Actions: Limit workflow run or job concurrency - GitHub Changelog</a>.
  </blockquote>
</aside>
